src/OpenLoco/Ui/Screenshot.cpp

```diff
@@ -117,6 +91,116 @@ namespace OpenLoco::Input
+    static std::string prepareSaveScreenshot(Gfx::RenderTarget& rt)
+    {
+        auto basePath = Platform::getUserDirectory();
+        std::string scenarioName = S5::getOptions().scenarioName;
+
+        if (scenarioName.length() == 0)
+            scenarioName = StringManager::getString(StringIds::screenshot_filename_template);
+
+        std::string fileName = std::string(scenarioName) + ".png";
+        fs::path path;
+        int16_t suffix;
+        for (suffix = 1; suffix < std::numeric_limits<int16_t>().max(); suffix++)
+        {
+            if (!fs::exists(basePath / fileName))
+            {
+                path = basePath / fileName;
+                break;
+            }
+
+            fileName = std::string(scenarioName) + " (" + std::to_string(suffix) + ").png";
+        }
+
+        if (path.empty())
+        {
+            throw std::runtime_error("Failed finding filename");
+        }
+
+        std::fstream outputStream(path.c_str(), std::ios::out | std::ios::binary);
+        saveRenderTargetToPng(rt, outputStream);
+
+        return fileName;
+    }
+
+    std::string saveScreenshot()
+    {
+        auto& rt = Gfx::getScreenRT();
+        return prepareSaveScreenshot(rt);
+    }
+
+    std::string saveGiantScreenshot()
+    {
+        const auto& main = WindowManager::getMainWindow();
+        const auto zoomLevel = main->viewports[0]->zoom;
+
+        const uint16_t resolutionWidth = ((Map::kMapColumns * 32 * 2) >> zoomLevel) + 8;
+        const uint16_t resolutionHeight = ((Map::kMapRows * 32 * 1) >> zoomLevel) + 128;
+
+        Ui::Viewport viewport{};
+        viewport.width = resolutionWidth;
+        viewport.height = resolutionHeight;
+        viewport.x = 0;
+        viewport.y = 0;
+        viewport.viewWidth = viewport.width;
+        viewport.viewHeight = viewport.height;
+        viewport.zoom = zoomLevel;
+        viewport.pad_11 = 0;
+        viewport.flags = 0;
+
+        const uint16_t centreX = (Map::kMapColumns / 2) * 32 + 16;
+        const uint16_t centreY = (Map::kMapRows / 2) * 32 + 16;
+
+        const uint16_t z = Map::TileManager::getHeight({ centreX, centreY }).landHeight;
+        const auto rotation = main->viewports[0]->getRotation();
+        uint16_t x = 0, y = 0;
+
+        // TODO: I think we have a function for this already?
+        switch (rotation)
+        {
+            case 0:
+                x = centreY - centreX;
+                y = ((centreX + centreY) / 2) - z;
+                break;
+            case 1:
+                x = -centreY - centreX;
+                y = ((-centreX + centreY) / 2) - z;
+                break;
+            case 2:
+                x = -centreY + centreX;
+                y = ((-centreX - centreY) / 2) - z;
+                break;
+            case 3:
+                x = centreY + centreX;
+                y = ((centreX - centreY) / 2) - z;
+                break;
+        }
+
+        viewport.viewX = x - ((viewport.viewWidth << zoomLevel) / 2);
```

Comment:
i dont think the zoomLevel should be here. Its all interms of  the viewport zoom these variables so it should just be viewX = x - viewWidth/ 2

Reply:
centre2dCoordinates would then make sense for this function. Oh wait i see the issue. Yes do `centre2dCoordinates`.